packages/esm-service-queues-app/src/queue-table/queue-entry-actions/queue-entry-actions.modal.tsx

```diff
@@ -287,89 +251,52 @@ export const QueueEntryActionModal: React.FC<QueueEntryActionModalProps> = ({
                   title={t('noPrioritiesConfigured', 'No priorities configured')}
                 />
               ) : (
-                <ContentSwitcher
-                  size="sm"
-                  selectedIndex={selectedPriorityIndex}
-                  onChange={(event) => {
-                    setSelectedPriorityUuid(event.name as string);
+                <RadioButtonGroup
+                  className={styles.radioButtonGroup}
+                  valueSelected={formState.selectedPriority}
+                  onChange={(uuid) => {
+                    setSelectedPriorityUuid(uuid);
                   }}>
                   {priorities?.map(({ uuid, display }) => (
-                    <Switch
-                      role="radio"
-                      name={uuid}
-                      text={
-                        uuid == queueEntry.priority.uuid
-                          ? t('currentValueFormatted', '{{value}} (Current)', {
-                              value: display,
-                              interpolation: { escapeValue: false },
-                            })
-                          : display
-                      }
+                    <RadioButton
                       key={uuid}
+                      name={display}
+                      labelText={
+                        <Tag
+                          className={classNames(styles.tag, {
+                            [styles.orange]: findPriorityIndex(uuid) === 1,
+                          })}
+                          role="radio"
+                          key={uuid}
+                          value={uuid}
+                          type={(() => {
+                            const index = findPriorityIndex(uuid);
+                            return index === 0 ? 'green' : index === 2 ? 'red' : '';
```

Comment:
```suggestion
                            // TODO: fix priority colors. https://openmrs.atlassian.net/browse/O3-4469
                            return index === 0 ? 'green' : index === 2 ? 'red' : '';
```

We can keep this for now, since we want to change it anyway. But... This assumes that the queue has 3 priorities in order of least urgent to most. This works for the queues in the refapp, but is not a safe assumption in general.